app.py

```diff
@@ -23,6 +23,10 @@
 
 class SetFrameRange(Application):
 
+    # declared here it is accesible to hooks without funky imports
+    class FrameOperationNotSupported(Exception):
```

Comment:
Where is this used? Is it needed?

Also the comment here maybe should be:
`# declared here so that it is accessible to hooks without funky imports.`